frontend/src/features/side_window/beacon_malfunctions/BeaconMalfunctionsBoard.tsx

```diff
@@ -223,25 +233,31 @@ export function BeaconMalfunctionsBoard() {
         sensors={sensors}
       >
         <Columns data-cy="side-window-beacon-malfunctions-columns" style={columnsStyle}>
-          {Object.keys(beaconMalfunctionsStages)
-            .filter(stage => beaconMalfunctionsStages[stage].isColumn)
-            .map((stageId, index) => (
-              <Droppable
-                key={stageId}
-                disabled={stageId === beaconMalfunctionsStages.END_OF_MALFUNCTION.code}
-                id={stageId}
-                index={index}
-              >
-                <StageColumn
-                  activeBeaconMalfunction={activeBeaconMalfunction}
-                  baseUrl={baseUrl}
-                  beaconMalfunctions={filteredBeaconMalfunctions[stageId] || []}
-                  isDroppedId={!!isDroppedId}
-                  stage={beaconMalfunctionsStages[stageId]}
-                  updateVesselStatus={updateVesselStatus}
-                />
-              </Droppable>
-            ))}
+          {Object.keys(beaconMalfunctionsStageColumnRecord)
```

Comment:
Not a big fan of "keys to maps", tranformations and/or filtering within the JSX, this is heavy without memoization and includes code logic within the rendering.

Reply:
Ok I'll move it to a `useMemo`, I agree for the code logic but this isn't really heavy here and I read that because of the hoverhead introduced by `useMemo`, we should avoid to use it by default on every variable. But let's discuss that next time !